node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/HostCapacityMaintainer.java

```diff
@@ -219,15 +220,19 @@ private List<Node> provisionHosts(int count, NodeResources nodeResources, Option
             }
             Version osVersion = nodeRepository().osVersions().targetFor(NodeType.host).orElse(Version.emptyVersion);
             List<Integer> provisionIndices = nodeRepository().database().readProvisionIndices(count);
+            HostSharing sharingMode = clusterType.map(ClusterSpec.Type::isContainer).orElse(false) ? HostSharing.exclusive : HostSharing.shared;
             HostProvisionRequest request = new HostProvisionRequest(provisionIndices, NodeType.host, nodeResources,
                                                                     ApplicationId.defaultId(), osVersion,
-                                                                    HostSharing.shared, clusterType, Optional.empty(),
+                                                                    sharingMode, clusterType, Optional.empty(),
                                                                     nodeRepository().zone().cloud().account(), false);
             List<Node> hosts = new ArrayList<>();
             hostProvisioner.provisionHosts(request,
                                            resources -> true,
                                            provisionedHosts -> {
-                                               hosts.addAll(provisionedHosts.stream().map(host -> host.generateHost(Duration.ZERO)).toList());
+                                               hosts.addAll(provisionedHosts.stream()
+                                                                            .map(host -> host.generateHost(Duration.ZERO))
+                                                                            .map(host -> host.withExclusiveToApplicationId(null))
```

Comment:
Throw an exception here or in withExclusiveToApplicationId() if provisionedForApplicationId is present?

Reply:
Yes, could just remove the nulling of provisionedFor in this setter. I originally wanted this as well.